pkg/api/controllers/settings/settings.go

```diff
@@ -98,7 +98,9 @@ func (s *settingsProvider) SetAll(settings map[string]settings.Setting) error {
 				obj.Default = setting.Default
 				update = true
 			}
-			if value != "" && obj.Value != value {
+			if value != "" && // if we get setting value from env, we might need to set it to setting object.
+				(!setting.KeepValue && obj.Value != value ||
+					setting.KeepValue && obj.Value == "") {
```

Comment:
I agree with what you are trying to accomplish (preserve value that is in DB after restart), but I don't agree with the KeepValue paradigm you have.

I think this should be the behavior for ALL settings and that we shouldnt make it a special case.

So, I think the logic should be this:

if obj.Value != "", we have a value that was set by the user. Keep it. do not override with the env var value (the value variable here)
else if value != "", we have an env var value set and it has not been overridden in the API, use the env var.
else, there is neither an env var or a db setting. do nothing.

Reply:
@vincent99 think we should continue with that paradigm in 2.0?

Reply:
@ibuildthecloud ^ ?

Reply:
Conversely, a setting can be marked un-editable if it was set by an environment var, so that's not confusing (if implemented)

Reply:
@cjellick The KeepValue paradigm can not be applied to all the settings because the `agent-image` still needs to be updated via environment variable.
@vincent99 If we want to keep the same paradigm as 1.x environment variable, we can not update `ui-index` in the API because we currently set ui-index in environment variable. It will need to have a new mechanism to update the default value for settings if we need something like 1.x.